ndb/tests/unit/test_model.py

```diff
@@ -2698,12 +2706,45 @@ class ThisKind(model.Model):
     )
     protobuf = helpers.entity_to_protobuf(datastore_entity)
     entity = model._entity_from_protobuf(protobuf)
-    assert type(entity) is ThisKind
+    assert isinstance(entity, ThisKind)
     assert entity.a == 42
     assert entity.b is None
     assert entity.c == gherkin
     assert entity.d == ["foo", "bar", "baz"]
     assert entity.e == [gherkin, dill]
+    assert entity._key.kind() == "ThisKind"
+    assert entity._key.id() == 123
```

Comment:
Because the mismatch between `app` and `project` was skeeving me out, because I don't really understand it, and this way I could work around not understanding that. But I've gone ahead and changed it. 

I'm not sure why datastore has "project" and ndb has "app", though. I'm guessing historical reasons.

Reply:
So we should probably migrate "app" to "project" as part of the port, then, right?